src/components/sign-in.js

```diff
@@ -1,59 +1,183 @@
+import {useForm} from "react-hook-form";
+import {useAuth} from "../context";
+import {useNavigate} from "react-router-dom";
+import {makeStyles} from "@material-ui/core/styles";
+import CssBaseline from "@material-ui/core/CssBaseline";
+import Avatar from "@material-ui/core/Avatar";
+import LockOutlinedIcon from "@material-ui/icons/LockOutlined";
+import Typography from "@material-ui/core/Typography";
+import Grid from "@material-ui/core/Grid";
+import InputField from "./InputField";
+import Container from "@material-ui/core/Container";
+import Box from "@material-ui/core/Box";
+import Link from "@material-ui/core/Link";
+import Button from "@material-ui/core/Button";
+import {IconButton, InputAdornment} from "@material-ui/core";
+import {Visibility, VisibilityOff} from "@material-ui/icons";
+import TextField from "@material-ui/core/TextField";
+import {Alert} from "@material-ui/lab";
+
+const useStyles = makeStyles((theme) => ({
+    paper: {
+        marginTop: theme.spacing(8),
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+    },
+    avatar: {
+        margin: theme.spacing(1),
+        backgroundColor: theme.palette.secondary.main,
+    },
+    form: {
+        width: '100%',
+        marginTop: theme.spacing(3),
+    },
+    submit: {
+        margin: theme.spacing(3, 0, 2),
+    },
+}));
+
+function Copyright() {
+    return (
+        <Typography variant="body2" color="textSecondary" align="center">
+            {'Copyright © '}
+            <Link color="inherit" href="https://github.com/CMSC-495-Group-One">
+                {/* Your Website */}
+                Git Repo
+            </Link>{' '}
+            {new Date().getFullYear()}
+            {'.'}
+        </Typography>
+    );
+}
+
+export default function SignInForm() {
+    const classes = useStyles();
+        const formMethods = useForm();
+        const { signIn } = useAuth();
+        const navigate = useNavigate();
+
+        const {
+            control,
+            register,
+            handleSubmit,
+            formState: { errors },
+        } = formMethods;
+
+        const [isLoading, setIsLoading] = useState(false);
+        const [loginError, setLoginError] = useState(null);
+
+
+        const onSubmit = async (data) => {
+            setIsLoading(true);
+            setLoginError(null);
+            try {
+                await signIn(data);
+                setIsLoading(false);
+                navigate("/tasks");
+            } catch (error) {
+                setIsLoading(false);
+                // handle error, e.g. show a message to the user
+                setLoginError("Invalid username or password.");
+            }
+        };
+
+    // Error Message Div:
+    const errorDiv =loginError
+        ? <Alert severity="error">{loginError}</Alert>
+        : '';
+
+    const [showPassword, setShowPassword] = useState(false);
+
+    const handleClickShowPassword = () => setShowPassword((show) => !show);
```

Comment:
Nice I din't know that the useState setter can accept a callback like this. Is `show` the current value of `showPassword`?

Reply:
Yes, that's the eye icon show or hide current password in the field.